vmdb/spec/models/miq_provision_virt_workflow_spec.rb

```diff
@@ -37,4 +37,61 @@
       end
     end
   end
+
+  context 'network selection' do
+    let(:workflow) { FactoryGirl.create(:miq_provision_virt_workflow) }
+
+    before do
+      FactoryGirl.create(:user_admin)
+      FactoryGirl.create(:miq_dialog_provision)
+      FactoryGirl.create(:small_environment)
+      FactoryGirl.create(:small_environment)
+      s11 = FactoryGirl.create(:switch, :name => "A")
+      s12 = FactoryGirl.create(:switch, :name => "B")
+      s13 = FactoryGirl.create(:switch, :name => "C")
+      @src_vm = VmOrTemplate.first
+      @host1  = Host.all[0]
+      @host2  = Host.all[1]
+      @src_vm.host.switches = [s11, s12, s13]
+      @lan11 = FactoryGirl.create(:lan, :name => "lan_A", :switch_id => s11.id)
+      @lan12 = FactoryGirl.create(:lan, :name => "lan_B", :switch_id => s12.id)
+      @lan13 = FactoryGirl.create(:lan, :name => "lan_C", :switch_id => s13.id)
+      @result1 = {'pg1' => ['switch1'],  'pg2' => ['switch2']}
+      @result2 = {'pg1' => ['switch21'], 'pg2' => ['switch2'], 'pg3' => ['switch23']}
+
+      @host1_dv_switches    = {'dvs_pg1' => 'pg1 (switch1)',
+                               'dvs_pg2' => 'pg2 (switch2)'}
+      @host2_dv_switches    = {'dvs_pg1' => 'pg1 (switch21)',
+                               'dvs_pg2' => 'pg2 (switch2)',
+                               'dvs_pg3' => 'pg3 (switch23)'}
+      @combined_dv_switches = {'dvs_pg1' => 'pg1 (switch1/switch21)',
+                               'dvs_pg2' => 'pg2 (switch2)',
+                               'dvs_pg3' => 'pg3 (switch23)'}
+
+      workflow.stub(:validate => true)
+      workflow.stub(:get_dialogs => {})
+      workflow.stub(:connect_to_ems).with(anything).and_return(nil)
+      workflow.stub(:get_host_dvs).with(@host1, nil).and_return(@result1)
+      workflow.stub(:get_host_dvs).with(@host2, nil).and_return(@result2)
+      workflow.instance_variable_set(:@values, :vm_tags => [], :src_vm_id => @src_vm.id)
+      workflow.instance_variable_set(:@target_resource, nil)
+    end
```

Comment:
@mkanoor Looks like there is a lot of room for improvement here with this test setup.  Many items setup here are only used in one test so they should be moved into a separate sub-context.  There is no need to create small_environment twice just to get a second host, just create the second host directly.  The *_dv_switches hashes can be pushed down into the individual tests where they are used and `@host2_dv_switches` is setup but never referenced.

Please take a pass at making this setup more efficient.